### PR TITLE
adding hdinsight cluster

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
Adding HDInsight cluster

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_hdinsightcluster

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_hdinsightcluster.py [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+        '''
+        self.log("Creating / Updating the Cluster instance {0}".format(self.name))
+
+        try:
+            if self.to_do == Actions.Create:
+                response = self.mgmt_client.clusters.create(resource_group_name=self.resource_group,
+                                                            cluster_name=self.name,
+                                                            parameters=self.parameters)
+            else:
+                response = self.mgmt_client.clusters.update(resource_group_name=self.resource_group,
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_hdinsightcluster.py]: Does this actually do anything (eg, are there any update operations that actually work on one of these)? By just passing RG/name, I'm guessing this is always a no-op (which is fine, but if it's a create/delete-only resource, might want to warn that updates do nothing...

File: lib/ansible/modules/cloud/azure/azure_rm_hdinsightcluster.py [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+
+    def delete_cluster(self):
+        '''
+        Deletes specified Cluster instance in the specified subscription and resource group.
+
+        :return: True
+        '''
+        self.log("Deleting the Cluster instance {0}".format(self.name))
+        try:
+            response = self.mgmt_client.clusters.delete(resource_group_name=self.resource_group,
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_hdinsightcluster.py:394]: should it block?

**Review by nitzmahone — CHANGES_REQUESTED** [review]
couple of minor things

File: lib/ansible/modules/cloud/azure/azure_rm_hdinsightcluster.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_hdinsightcluster.py]: indeed. should be possible to add tags, but seems like I missed it.... :-(
I will add a separate resize cluster functionality as well.

**Comment by Fred-sun** [comment]
@yungezz  @yuwzho

**Comment by Fred-sun** [comment]
@yuwzho @yungezz review_need

**Review by nitzmahone — APPROVED** [review]

**Comment by mattclay** [comment]
@zikalino I marked this test unstable in https://github.com/ansible/ansible/pull/54152 due to tests timing out when executed as a group. I didn't want to mark it unsupported, but I didn't really want to assign it to its own test group either.

**Comment by zikalino** [comment]
@mattclay I think it's a good choice
I think we should have some special alias for resources like this, so we could have enabled tests, but in the same time we shouldn't run them too frequently.
another example is aks which is disabled right now.

**Pull request merged**